zebra-chain/src/transaction/serialize.rs

```diff
@@ -377,16 +379,20 @@ impl ZcashSerialize for orchard::ShieldedData {
         // Denoted as `bindingSigOrchard` in the spec.
         self.binding_sig.zcash_serialize(&mut writer)?;
 
+        #[cfg(feature = "tx-v6")]
+        // Denoted as `vAssetBurn` in the spec (ZIP 230).
+        self.burn.zcash_serialize(&mut writer)?;
+
         Ok(())
     }
 }
 
 // we can't split ShieldedData out of Option<ShieldedData> deserialization,
 // because the counts are read along with the arrays.
-impl ZcashDeserialize for Option<orchard::ShieldedData> {
+impl<V: OrchardFlavorExt> ZcashDeserialize for Option<orchard::ShieldedData<V>> {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
         // Denoted as `nActionsOrchard` and `vActionsOrchard` in the spec.
-        let actions: Vec<orchard::Action> = (&mut reader).zcash_deserialize_into()?;
+        let actions: Vec<orchard::Action<V>> = (&mut reader).zcash_deserialize_into()?;
```

Comment:
This method returns `None` `if actions.is_empty()`, but it likely needs to check for asset burns as well.